### PR TITLE
Update travis matrix, core patches and readme

### PR DESCRIPTION
**Pull request by golenkovm**
This is the fix for #71 

- [x] Refactor Travis matrix.
- [x] Add core patches for stable version (currently 31 and 27).
- [x] Update readme file and provide instructions how to apply patches.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -8,97 +8,72 @@ cache:
...
-  - DB=mysqli MOODLE_BRANCH=MOODLE_38_STABLE
-  - DB=pgsql MOODLE_BRANCH=master
-  - DB=mysqli MOODLE_BRANCH=master
-
 matrix:
-  exclude:
+  include:
     - php: 7.2
-      env: DB=pgsql MOODLE_BRANCH=MOODLE_31_STABLE_FSAPI MOODLE_REPO=https://github.com/kenneth-hendricks/moodle-fs-api.git
+      env: DB=mysqli MOODLE_BRANCH=master
```
Comment (brendanheywood) [inline .travis.yml:19]: this is a tiny dumb personal preference but can you put master down the bottom as it is closer to 3.8 and soon 3.9

File: .travis.yml [hunk 2]
```diff
@@ -8,97 +8,72 @@ cache:
...
-      env: DB=mysqli MOODLE_BRANCH=master
+    - php: 7.2
+      env: DB=pgsql  MOODLE_BRANCH=MOODLE_38_STABLE
+    - php: 7.1
+      env: DB=mysqli MOODLE_BRANCH=MOODLE_38_STABLE
+    - php: 7.1
+      env: DB=pgsql  MOODLE_BRANCH=MOODLE_38_STABLE
 
 before_install:
+  - export IGNORE_PATHS=classes/azure/StreamWrapper.php,classes/local/store/azure/stream_wrapper.php,classes/local/store/swift/stream_wrapper.php
```
Comment (brendanheywood) [inline .travis.yml:76]: What is this for?

File: README.md [hunk 3]
```diff
@@ -294,6 +294,55 @@ Openstack Object Storage settings
 - **Project ID**: Openstack project ID
 - **Container**: Name of the storage container
 
+## Applying core patches
+
+This plugin requires various trackers to be backported to maintain the plugin functionality.
+
+| Moodle version   | Patches |
+|------------------|---------|
+| Moodle 3.1       | [MDL-46375](https://tracker.moodle.org/browse/MDL-46375), [MDL-58068](https://tracker.moodle.org/browse/MDL-58068) |
```
Comment (brendanheywood) [inline README.md:303]: can you please flesh this out with version ranges like 3.1 - 3.6 so we cover everything explicitly

File: README.md [hunk 4]
```diff
@@ -294,6 +294,55 @@ Openstack Object Storage settings
...
+| Moodle 2.7       | [MDL-46375](https://tracker.moodle.org/browse/MDL-46375), [MDL-58068](https://tracker.moodle.org/browse/MDL-58068), [MDL-49627](https://tracker.moodle.org/browse/MDL-49627) |
+
+#### Moodle 3.1:
```
Comment (brendanheywood) [inline README.md:306]: make these headings match the ranges above ie 3.1 - 3.x

File: .travis.yml (hunk 1 above)
Comment (golenkovm) [reply .travis.yml:19]: My logic was to have the list that we will append when new version releases and not adding new lines in the miidle of the list

Comment (golenkovm) [reply .travis.yml]: Moved master down the list

File: .travis.yml (hunk 2 above)
Comment (golenkovm) [reply .travis.yml]: Added `admin/tool/objectfs/thirdpartylibs.xml` instead

File: README.md (hunk 3 above)
Comment (golenkovm) [reply README.md]: Done

File: README.md (hunk 4 above)
Comment (golenkovm) [reply README.md:322]: Done

**Pull request merged**